test/Dialect/HW/hw-inliner.mlir

```diff
@@ -0,0 +1,261 @@
+hw.module private @OuterModule(in %a: i4, out b: i4) {
+  %0 = hw.instance "inner" @InnerModule(x: %a: i4) -> (y: i4)
+  hw.output %0 : i4
+}
+// CHECK-NOT: hw.module private @InnerModule
+hw.module private @InnerModule(in %x: i4, out y: i4) {
+  %0 = comb.add %x, %x : i4
+  %1 = comb.mul %0, %0 : i4
+  hw.output %1 : i4
+}
+
+// Test multiple outputs
+// CHECK-LABEL: hw.module @InlineMultiOutput
+hw.module @InlineMultiOutput(in %x: i4, out y: i4, out z: i4) {
+  // CHECK-NOT: hw.instance
+  // CHECK-NEXT: %[[V0:.+]] = comb.add %x, %x
+  // CHECK-NEXT: %[[V1:.+]] = comb.mul %x, %x
+  // CHECK-NEXT: hw.output %[[V0]], %[[V1]]
+  %0, %1 = hw.instance "multi_out" @MultiOutputModule(a: %x: i4) -> (b: i4, c: i4)
+  hw.output %0, %1 : i4, i4
+}
+// CHECK-NOT: hw.module private @MultiOutputModule
+hw.module private @MultiOutputModule(in %a: i4, out b: i4, out c: i4) {
+  %0 = comb.add %a, %a : i4
+  %1 = comb.mul %a, %a : i4
+  hw.output %0, %1 : i4, i4
+}
+
+// Test inner symbol renaming with multiple instances
+// CHECK-LABEL: hw.module @InlineMultiInstanceInnerSym
+hw.module @InlineMultiInstanceInnerSym(in %x: i4, in %y: i4, out a: i4, out b: i4) {
+  // CHECK-NOT: hw.instance
+  // CHECK-DAG: %{{.+}} = hw.wire %{{.+}} sym @wire_sym{{(_[0-9]+)?}}
+  // CHECK-DAG: %{{.+}} = hw.wire %{{.+}} sym @wire_sym{{(_[0-9]+)?}}
+  // CHECK: hw.output
+  %0 = hw.instance "inst1" @ModuleWithSym(a: %x: i4) -> (b: i4)
+  %1 = hw.instance "inst2" @ModuleWithSym(a: %y: i4) -> (b: i4)
+  hw.output %0, %1 : i4, i4
+}
+// CHECK-NOT: hw.module private @ModuleWithSym
+hw.module private @ModuleWithSym(in %a: i4, out b: i4) {
+  %0 = comb.add %a, %a : i4
+  %1 = hw.wire %0 sym @wire_sym : i4
+  hw.output %1 : i4
+}
+
+// Test inner symbol renaming with multiple symbols in same module
+// CHECK-LABEL: hw.module @InlineMultipleInnerSyms
+hw.module @InlineMultipleInnerSyms(in %x: i4, in %y: i4, out a: i4, out b: i4) {
+  // CHECK-NOT: hw.instance
+  // CHECK-DAG: %{{.+}} = hw.wire %{{.+}} sym @wire1{{(_[0-9]+)?}}
+  // CHECK-DAG: %{{.+}} = hw.wire %{{.+}} sym @wire2{{(_[0-9]+)?}}
+  // CHECK-DAG: %{{.+}} = hw.wire %{{.+}} sym @wire1{{(_[0-9]+)?}}
+  // CHECK-DAG: %{{.+}} = hw.wire %{{.+}} sym @wire2{{(_[0-9]+)?}}
+  // CHECK: hw.output
+  %0 = hw.instance "inst1" @ModuleWithMultipleSyms(a: %x: i4) -> (b: i4)
+  %1 = hw.instance "inst2" @ModuleWithMultipleSyms(a: %y: i4) -> (b: i4)
+  hw.output %0, %1 : i4, i4
+}
+// CHECK-NOT: hw.module private @ModuleWithMultipleSyms
+hw.module private @ModuleWithMultipleSyms(in %a: i4, out b: i4) {
+  %0 = comb.add %a, %a : i4
+  %1 = hw.wire %0 sym @wire1 : i4
+  %2 = comb.mul %1, %a : i4
+  %3 = hw.wire %2 sym @wire2 : i4
+  hw.output %3 : i4
+}
+
+// Test InnerRefAttr updating with single instance
+// CHECK-LABEL: hw.module @InlineWithInnerRef
+hw.module @InlineWithInnerRef(in %x: i4, out y: i4) {
+  // CHECK-NOT: hw.instance
+  // CHECK-NEXT: %[[V0:.+]] = comb.add %x, %x
+  // CHECK-NEXT: %[[W0:.+]] = hw.wire %[[V0]] sym @[[SYM:wire_op]]
```

Comment:
Nit: there isn't any renaming here, so capturing this into `SYM` doesn't serve a purpose. The test probably does want to not capture `wire_op` exactly and should be `.+` or an alpha-numeric capture.